lib/inferno/dsl/fhir_evaluation/rules/all_references_resolve.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllReferencesResolve < Rule
+          def check(context)
+            # resource_type_ids is for quick look up when there is a reference type
+            # resource_ids is for quick look up when there is no type (i.e. uuid used)
+            util = Inferno::Utils::EvaluatorUtil
+            _, resource_type_ids, resource_ids, references = util.extract_ids_references(context.data)
+            unresolved_references = Hash.new { |h, k| h[k] = [] }
+            references.each do |k, v|
+              v.each do |reference|
+                # no type for the reference
+                if reference[1] == ''
+                  unresolved_references[k] << reference unless resource_ids.include?(reference[2])
+                elsif !resource_type_ids[reference[1]].include?(reference[2])
+                  unresolved_references[k] << reference
+                end
+              end
+            end
+
+            if unresolved_references.any?
+              message = gen_reference_fail_message(unresolved_references)
+              result = EvaluationResult.new(message, rule: self)
+            else
+              message = 'All references resolve'
+              result = EvaluationResult.new(message, severity: 'success', rule: self)
+            end
+
+            context.add_result result
+          end
+
+          def gen_reference_fail_message(unresolved_references)
+            "Found unresolved references: #{
+                            unresolved_references.map do |k, v|
+                              "\n Resource (id): #{k}  #{v.each_with_index.map do |val, _idx|
```

Comment:
Don't do nested maps inside of a string interpolation.

Reply:
This has not been addressed. You shouldn't have multiline expressions inside of a string interpolation.

Reply:
Oh, I missed another remained in there. Fixed: https://github.com/inferno-framework/inferno-core/pull/583/commits/f8af437f8733cd446ec779779ef19046610bae8d

Reply:
Modified: https://github.com/inferno-framework/inferno-core/pull/583/commits/f8af437f8733cd446ec779779ef19046610bae8d